### DISCUSSION TITLE
Using Grid and  grid-area, is that possible?

### DISCUSSION THREAD

**Discussion by Devotee007** [tailwindlabs/tailwindcss#8039] [Help]
I'm about to do at three column to a two column using grid. The first column is supposed to fill up with items from bottom to top and then continue in the next column. So it's 5, 5, 4 items in the three column and 7, 7 in the two column. I have started to do it with Tailwind, but I got stuck at grid-area. I can't find it in the docs? Is there another way to do this?  

Edit: I forgot, the change from 3 to 2 columns shall happen on small screens. But I guess that's easy to do as always. :)

With classic CSS. I got this:

```
3 columns:

.parent {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(5, 1fr);
  grid-column-gap: 0px;
  grid-row-gap: 0px;
}


  .div1 { grid-area: 1 / 1 / 2 / 2; }
  .div2 { grid-area: 2 / 1 / 3 / 2; }
  .div3 { grid-area: 3 / 1 / 4 / 2; }
  .div4 { grid-area: 4 / 1 / 5 / 2; }
  .div5 { grid-area: 5 / 1 / 6 / 2; }
  .div6 { grid-area: 1 / 2 / 2 / 3; }
  .div7 { grid-area: 2 / 2 / 3 / 3; }
  .div8 { grid-area: 3 / 2 / 4 / 3; }
  .div9 { grid-area: 4 / 2 / 5 / 3; }
  .div10 { grid-area: 5 / 2 / 6 / 3; }
  .div11 { grid-area: 1 / 3 / 2 / 4; }
  .div12 { grid-area: 2 / 3 / 3 / 4; }
  .div13 { grid-area: 3 / 3 / 4 / 4; }
  .div14 { grid-area: 4 / 3 / 5 / 4; }
```

```
2 columns:

.parent {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(7, 1fr);
  grid-column-gap: 0px;
  grid-row-gap: 0px;
  }

.div1 { grid-area: 1 / 1 / 2 / 2; }
.div2 { grid-area: 2 / 1 / 3 / 2; }
.div3 { grid-area: 3 / 1 / 4 / 2; }
.div4 { grid-area: 4 / 1 / 5 / 2; }
.div5 { grid-area: 5 / 1 / 6 / 2; }
.div6 { grid-area: 6 / 1 / 7 / 2; }
.div7 { grid-area: 7 / 1 / 8 / 2; }
.div8 { grid-area: 1 / 2 / 2 / 3; }
.div9 { grid-area: 2 / 2 / 3 / 3; }
.div10 { grid-area: 3 / 2 / 4 / 3; }
.div11 { grid-area: 4 / 2 / 5 / 3; }
.div12 { grid-area: 5 / 2 / 6 / 3; }
.div13 { grid-area: 6 / 2 / 7 / 3; }
.div14 { grid-area: 7 / 2 / 8 / 3; }

```
```
The HTML is like this:

   <div class="parent">
    <div class="div1">1 </div>
    <div class="div2">2 </div>
    <div class="div3">3 </div>
    <div class="div4">4 </div>
    <div class="div5">5 </div>
    <div class="div6">6 </div>
    <div class="div7">7 </div>
    <div class="div8">8 </div>
    <div class="div9">9 </div>
    <div class="div10">10 </div>
    <div class="div11">11 </div>
    <div class="div12">12 </div>
    <div class="div13">13 </div>
    <div class="div14">14 </div>
  </div>
```

**Comment by horuskol** [comment]
Tailwind CSS core doesn't have utilities for grid-area or grid-template-areas - so I made a plugin to help with this: https://www.npmjs.com/package/@savvywombat/tailwindcss-grid-areas

However, I'm not sure that grid-area is the best option - there's a good chance of making a mistake with all those very similar values.

Working from the top-left is trivial: `<div class="grid grid-cols-[repeat(3,_3rem)] grid-rows-[repeat(5,_3rem)] grid-flow-col">`

But if you need to work from the bottom-left, there's no way to set the direction for filling the grid. You might be better off with flex - if your parent height is constrained you can set the height of your child elements with something like `h-[calc(100%/5)]`, and flex will wrap after every fifth item.
